### DISCUSSION TITLE
vpnclound listen two internet interfaces

### DISCUSSION THREAD

**Discussion by biolim** [dswd/vpncloud#190] [Q&A]
Is it possible to run and manage connections on linux with 2 wan interfaces? Will it work? Perhaps you should run 2 daemons and collect them in active-passive bonding!?

**Comment by dswd** [comment]
(moved from issues)

I am not sure I understand you correctly. Connections in VpnCloud happen on the network socket side on layer 4 (UDP) while bonding on network interfaces would be on the virtual device side on layer 2. So I hope my answer still helps:

VpnCloud will only create one virtual device per instance. You can use a bridge to connect it to other devices on layer 2 (Ethernet). You will only be able to attach it to one bridge at a time and if you want to connect it to 2 other devices you would need to put all 3 devices on the same bridge which would also cross-connect those 2 devices.
You can run 2 instances on the same node to have 2 devices that you can attach to 2 different bridges. However you should be careful to avoid broadcast loops as VpnCloud does not protect you from this.
On layer 3 (IP) you are much more flexible. You can work with one instance, enable forwarding and use the full power of routing.

If you are talking about the UDP connections between VpnCloud instances, they don't care about network devices at all. VpnCloud just uses normal routing to reach its peers, no matter which interface that uses. (Yes you have to avoid to recursively route the VpnCloud traffic over VpnCloud).
